File: model/src/main/java/org/bandrsoftwares/celestialdiary/model/dto/establishment/EstablishmentBundleDTO.java
package org.bandrsoftwares.celestialdiary.model.dto.establishment;

import org.bandrsoftwares.celestialdiary.model.dto.saleable.bundle.WrappedBundleDTO;
import org.bandrsoftwares.celestialdiary.model.mongodb.establishment.EstablishmentBundle;

public record EstablishmentBundleDTO(double customPrice, boolean usingCustomPrice, WrappedBundleDTO bundle) {

    public EstablishmentBundleDTO(EstablishmentBundle establishmentBundle) {
        this(establishmentBundle.getCustomPrice(), establishmentBundle.getUsingCustomPrice(), new WrappedBundleDTO(establishmentBundle.getBundle()));
    }
}

File: ui-frontend/src/data/model/establishment/establishment-bundle-dto.ts
import {EstablishmentSaleableDTO} from "./establishment-saleable-dto";
import {WrappedBundleDTO} from "../saleable/bundle/wrapped-bundle-dto";

export interface EstablishmentBundleDTO extends EstablishmentSaleableDTO {
  bundle: WrappedBundleDTO
}
